--- pari-2.1.5/config/arch-osname.orig	2004-03-05 19:41:45.000000000 +0100
+++ pari-2.1.5/config/arch-osname	2004-03-05 20:09:37.036819216 +0100
@@ -16,7 +16,7 @@
   osf1)     case "$5" in alpha) arch=alpha;; esac;;
   cygwin*)  arch=$HOSTTYPE; osname=cygwin;;
   linux)    arch=`uname -m | sed s/i.86/ix86/`
-	    case "`cat /proc/cpuinfo`" in
+	    case "generic" in
 	    *SuperSparc*)   arch=sparcv8_super;;
 	    *TMS390Z5[05]*) arch=sparcv8_super;; # SuperSparc I or II
 	    *TMS390S1[05]*) arch=sparcv8_micro;; # MicroSparc I
--- pari-2.1.7/gp2c-0.0.4pl2/config/arch-osname.orig	2001-03-18 21:55:19.000000000 +0100
+++ pari-2.1.7/gp2c-0.0.4pl2/config/arch-osname	2004-03-05 20:09:50.367792600 +0100
@@ -16,7 +16,7 @@
   osf1)     case "$5" in alpha) arch=alpha;; esac;;
   cygwin*)  arch=$HOSTTYPE;;
   linux)    arch=`uname -m`
-	    case "`cat /proc/cpuinfo`" in
+	    case "generic" in
 	    *SuperSparc*)   arch=sparcv8_super;;
 	    *TMS390Z5[05]*) arch=sparcv8_super;; # SuperSparc I or II
 	    *TMS390S1[05]*) arch=sparcv8_micro;; # MicroSparc I
--- pari-2.1.7/Math-Pari-2.010804/utils/Math/PariBuild.pm.orig	2005-01-20 03:04:46.000000000 +0100
+++ pari-2.1.7/Math-Pari-2.010804/utils/Math/PariBuild.pm	2005-01-22 00:01:08.135658528 +0100
@@ -801,7 +801,7 @@
     chomp($machine = `uname -m`);
     $machine = 'sparcv9' if $machine eq 'sparc64';
     $machine = 'hppa' if $machine =~ /^parisc/;
-    if ($machine eq 'sparc' and -e '/proc/cpuinfo' && -R '/proc/cpuinfo') {
+    if ($machine eq 'sparc' and 0 && -e '/proc/cpuinfo' && -R '/proc/cpuinfo') {
       open IN, '/proc/cpuinfo' or die "open /proc/cpuinfo: $!";
       local $/ = undef;		# Needed?
       my $info = <IN>;
