--- pari-2.3.5/config/arch-osname~	2009-09-19 14:01:14.000000000 +0200
+++ pari-2.3.5/config/arch-osname	2010-06-12 18:53:11.328315316 +0200
@@ -34,7 +34,7 @@
   darwin*)  arch=`uname -p`
             if test "$arch" = powerpc; then arch=ppc; fi ;;
   osf1)     case "$5" in alpha) arch=alpha;; esac;;
-  linux)    arch=`uname -m`
+  linux)    arch=`uname -m | sed s/i.86/ix86/`
 	    case $arch in
 		sparc64) arch=sparcv9;;
 		parisc*) arch=hppa;;
