diff -urN pari-2.1.1/Configure pari-2.1.1.patch/Configure
--- pari-2.1.1/Configure	Fri Apr 27 15:51:44 2001
+++ pari-2.1.1.patch/Configure	Thu May 10 12:55:15 2001
@@ -1145,7 +1145,7 @@
   dfltall=$prefix/lib/pari
 else
   dfltman=$share_prefix/man/man1
-  dfltall=$share_prefix/doc/pari
+  dfltall=$share_prefix/parigp
 fi
 
 echo $n ..."\"gp\" executable ? $c"
@@ -1166,7 +1166,7 @@
 if test -z "$share_prefix"; then
   dfltdat=$miscdir/galdata
 else
-  dfltdat=$share_prefix/pari/galdata
+  dfltdat=$share_prefix/parigp/galdata
 fi
 
 echo $n ..."miscellaneous data (galois resolvents) ? $c"
