diff -urN pari-2.1.1/src/language/init.c pari-2.1.1.patch/src/language/init.c
--- pari-2.1.1/src/language/init.c	Mon Mar 26 16:11:43 2001
+++ pari-2.1.1.patch/src/language/init.c	Mon May 21 13:39:09 2001
@@ -1460,6 +1460,7 @@
 
 # include <sys/times.h>
 # include <sys/time.h>
+# include <time.h>
   static long
   timer_proto(int i)
   {
@@ -1468,7 +1469,7 @@
     long delay;
 
     times(&t);
-    delay = (long)((t.tms_utime - oldticks[i]) * (1000. / CLK_TCK));
+    delay = (long)((t.tms_utime - oldticks[i]) * (1000. / CLOCKS_PER_SEC));
     oldticks[i] = t.tms_utime;
     return delay;
   }
