--- pari-2.1.5/config/arch-osname.orig	2004-03-05 19:41:45.000000000 +0100
+++ pari-2.1.5/config/arch-osname	2004-03-05 20:09:37.036819216 +0100
@@ -16,7 +16,7 @@
 	    case $arch in
 		sparc64) arch=sparcv9;;
 		parisc*) arch=hppa;;
-		sparc) case "`cat /proc/cpuinfo`" in
+		sparc) case "generic" in
 	    *SuperSparc*)   arch=sparcv8_super;;
 	    *TMS390Z5[05]*) arch=sparcv8_super;; # SuperSparc I or II
 	    *TMS390S1[05]*) arch=sparcv8_micro;; # MicroSparc I
