diff -urN pari-2.1.0/Configure pari-2.1.0-FHS/Configure
--- pari-2.1.0/Configure	Mon Nov 13 15:35:43 2000
+++ pari-2.1.0-FHS/Configure	Sat Dec  2 20:25:25 2000
@@ -1153,10 +1153,10 @@
 dflt=$prefix/include/pari; . ./myread; includedir=$ans
 
 echo $n ..."manual pages ? $c"
-dflt=$prefix/man/man1; . ./myread; mandir=$ans
+dflt=$prefix/share/man/man1; . ./myread; mandir=$ans
 
 echo $n ..."documentation, help scripts, and emacs macros ? $c"
-dflt=$prefix/lib/pari; . ./myread; miscdir=$ans
+dflt=$prefix/share/parigp; . ./myread; miscdir=$ans
 
 echo $n ..."miscellaneous data (galois resolvents) ? $c"
 dflt=$miscdir/data; . ./myread; datadir=$ans
