--- pari-2.1.6/src/kernel/sparcv8/Makefile.SH.orig	2002-06-30 20:26:53.000000000 +0200
+++ pari-2.1.6/src/kernel/sparcv8/Makefile.SH	2004-03-05 19:41:45.007006384 +0100
@@ -26,9 +26,9 @@
 then
 cat >> $file << EOT
 kernel1.s: $kernel1
-	\$(CPP) $KERNELCPPFLAGS $kernel1 > \$@
+	\$(CPP) $KERNELCPPFLAGS - < $kernel1 > \$@
 kernel2.s: $kernel2
-	\$(CPP) $KERNELCPPFLAGS $kernel2 > \$@
+	\$(CPP) $KERNELCPPFLAGS - < $kernel2 > \$@
 EOT
 kernel1=kernel1.s
 kernel2=kernel2.s
--- pari-2.1.5/src/kernel/sparcv8/level0.h.orig	2000-11-03 22:00:27.000000000 +0100
+++ pari-2.1.5/src/kernel/sparcv8/level0.h	2004-03-05 19:56:44.537257000 +0100
@@ -64,7 +64,7 @@
           addx  %%g0,%%g0,%1" \
 	 : "=r" (__value), "=r" (overflow) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g6","cc"); \
+         : "cc"); \
 __value; })								
 
 #define addllx(a,b) \
@@ -74,7 +74,7 @@
           addx  %%g0,%%g0,%1" \
 	 : "=r" (__value), "=r" (overflow) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g6","cc"); \
+         : "cc"); \
 __value; })								
 
 #define subll(a,b) \
@@ -83,7 +83,7 @@
           addx  %%g0,%%g0,%1" \
 	 : "=r" (__value), "=r" (overflow) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g6","cc"); \
+         : "cc"); \
 __value; })								
 
 #define subllx(a,b) \
@@ -93,7 +93,7 @@
           addx  %%g0,%%g0,%1" \
 	 : "=r" (__value), "=r" (overflow) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g6","cc"); \
+         : "cc"); \
 __value; })								
 
 #define shiftl(a,b) \
@@ -103,7 +103,7 @@
           sll %2,%3,%0" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%o4","%g5"); \
+         : "%o4"); \
 __value; })								
 
 #define shiftl2(a,b,c) \
@@ -112,7 +112,7 @@
           sll %2,%3,%0" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2), "r" (__arg3) \
-         : "%g5"); \
+         ); \
 __value; }) 			
 
 #define shiftlr(a,b) \
@@ -122,7 +122,7 @@
           srl %2,%3,%0" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%o4","%g5"); \
+         : "%o4"); \
 __value; }) 			
 
 #define shiftlr1(a) \
@@ -131,7 +131,7 @@
           srl %2,1,%0" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1) \
-         : "%g5"); \
+         ); \
 __value; }) 			
 
 #define shiftlr2(a,b,c) \
@@ -140,7 +140,7 @@
           srl %2,%3,%0" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2), "r" (__arg3) \
-         : "%g5"); \
+         ); \
 __value; }) 			
 
 #define mulll(a,b) \
@@ -149,7 +149,7 @@
           rd  %%y,%1" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g5");	\
+         );	\
 __value;})								
 
 #define addmul(a,b) \
@@ -160,7 +160,7 @@
           addx %%g0,%%o4,%1" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%o4","%g5","cc");	\
+         : "%o4","cc");	\
 __value;})								
 
 #define divllasm(a,b) \
@@ -176,7 +176,7 @@
 1:	 sub     %%o4,%%o5,%1"\
 	: "=r" (__value), "=r" (hiremainder) \
 	: "r" (__arg1), "r" (__arg2) \
-        : "%o4","%o5","%g5","cc");	\
+        : "%o4","%o5","cc");	\
 __value;})								
 
 #endif /* ASMINLINE */
--- pari-2.1.5/src/kernel/sparcv9/asmsparcv9.h.orig	2000-11-03 22:00:27.000000000 +0100
+++ pari-2.1.5/src/kernel/sparcv9/asmsparcv9.h	2004-03-05 19:57:55.121526560 +0100
@@ -25,7 +25,7 @@
           addc  %%g0,%%g0,%1" \
 	 : "=r" (__value), "=r" (overflow) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g6","cc"); \
+         : "cc"); \
 __value; })								
 
 #define subll(a,b) \
@@ -34,7 +34,7 @@
           addc  %%g0,%%g0,%1" \
 	 : "=r" (__value), "=r" (overflow) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g6","cc"); \
+         : "cc"); \
 __value; })								
 
 #define addllx(a,b) \
@@ -44,7 +44,7 @@
           addc  %%g0,%%g0,%1" \
 	 : "=r" (__value), "=r" (overflow) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g6","cc"); \
+         : "cc"); \
 __value; })								
 
 #define subllx(a,b) \
@@ -54,7 +54,7 @@
           addc  %%g0,%%g0,%1" \
 	 : "=r" (__value), "=r" (overflow) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g6","cc"); \
+         : "cc"); \
 __value; })								
 
 #define shiftl(a,b) \
@@ -64,7 +64,7 @@
           sll %2,%3,%0" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%o4","%g5"); \
+         : "%o4"); \
 __value; })								
 
 #define shiftl2(a,b,c) \
@@ -73,7 +73,7 @@
           sll %2,%3,%0" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2), "r" (__arg3) \
-         : "%g5"); \
+         ); \
 __value; }) 			
 
 #define shiftlr(a,b) \
@@ -83,7 +83,7 @@
           srl %2,%3,%0" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%o4","%g5"); \
+         : "%o4"); \
 __value; }) 			
 
 #define shiftlr1(a) \
@@ -92,7 +92,7 @@
           srl %2,1,%0" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1) \
-         : "%g5"); \
+         ); \
 __value; }) 			
 
 #define shiftlr2(a,b,c) \
@@ -101,7 +101,7 @@
           srl %2,%3,%0" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2), "r" (__arg3) \
-         : "%g5"); \
+         ); \
 __value; }) 			
 
 #define mulll(a,b) \
@@ -110,7 +110,7 @@
           srlx %0,32,%1" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g5");	\
+         );	\
 __value;})								
 
 #define addmul(a,b) \
@@ -120,7 +120,7 @@
           srlx %0,32,%1" \
 	 : "=r" (__value), "=r" (hiremainder) \
 	 : "r" (__arg1), "r" (__arg2) \
-         : "%g5");	\
+         );	\
 __value;})								
 
 #define divll(a,b) \
@@ -136,7 +136,7 @@
 1:	 sub     %%o4,%%o5,%1"\
 	: "=r" (__value), "=r" (hiremainder) \
 	: "r" (__arg1), "r" (__arg2) \
-        : "%o4","%o5","%g5","cc");	\
+        : "%o4","%o5","cc");	\
 __value;})								
 
 #endif
