--- pari-2.1.4/src/kernel/sparcv8/Makefile.SH.orig	Wed Feb 23 19:44:24 2000
+++ pari-2.1.4/src/kernel/sparcv8/Makefile.SH	Thu Sep  5 14:15:04 2002
@@ -17,7 +17,7 @@
 kernel1=$ker/level0_$asmarch.S
 kernel2=$ker/level0.S
 
-if test "$osname" = "nextstep" -o "$osname" = "linux"
+if test "$osname" = "nextstep"
 then
   level0=$src/kernel/none/asm0.h
 else
@@ -29,9 +29,9 @@
 then
 cat >> $file << EOT
 kernel1.s: $kernel1
-	\$(CPP) $KERNELCPPFLAGS $kernel1 > \$@
+	\$(CPP) $KERNELCPPFLAGS - < $kernel1 > \$@
 kernel2.s: $kernel2
-	\$(CPP) $KERNELCPPFLAGS $kernel2 > \$@
+	\$(CPP) $KERNELCPPFLAGS - < $kernel2 > \$@
 EOT
 kernel1=kernel1.s
 kernel2=kernel2.s
