diff -urNp -x '*.orig' pari-2.5.5.org/config/get_archos pari-2.5.5/config/get_archos
--- pari-2.5.5.org/config/get_archos	2011-09-22 22:02:11.000000000 +0200
+++ pari-2.5.5/config/get_archos	2021-05-07 23:08:59.794747185 +0200
@@ -6,6 +6,10 @@ fi
 arch=`echo "$target_host" | sed -e 's/\(.*\)-.*/\1/'`
 osname=`echo "$target_host" | sed -e 's/.*-\(.*\)/\1/'`
 
+if test -n "$target_arch"; then 
+  arch=$target_arch
+fi
+
 if test "$fastread" != yes; then
   cat << EOM
 ==========================================================================
--- pari-2.11.4/config/get_config_options.orig	2021-06-26 09:48:23.206303784 +0200
+++ pari-2.11.4/config/get_config_options	2021-06-26 09:50:09.378489327 +0200
@@ -48,6 +48,7 @@ while test $# -gt 0; do
   --sysdatadir=*) dfltsysdatadir=`echo "$1" | sed -e 's/[-a-z]*=//'`;;
   --time=*) timing_fun=`echo "$1" | sed -e 's/[-a-z]*=//'`;;
   --host=*) target_host=`echo "$1" | sed -e 's/[-a-z]*=//'`;;
+  --target=*) target_arch=`echo "$1" | sed -e 's/[-a-z]*=//'`;;
   --kernel=*) kernel=`echo "$1" | sed -e 's/[-a-z]*=//'`;;
   --mt=*) thread_engine=`echo "$1" | sed -e 's/[-a-z]*=//'`;;
   --tune) tune=yes;;
