--- pari-2.1.4/Configure.orig	Wed Sep 11 23:09:56 2002
+++ pari-2.1.4/Configure	Wed Sep 11 23:13:37 2002
@@ -278,7 +278,7 @@
 Currently supported architectures:
 EOM
   rep='none sparcv7 sparcv8_super sparcv8_micro sparcv9
-       m68k ix86 i386 i486 i586 i686 hppa alpha mips fx2800'
+       m68k ix86 i386 i486 i586 i686 athlon hppa alpha mips fx2800'
   . ./display
   echo $n ..."Which of these apply, if any ? $c"
   dflt=$arch; . ./myread; arch=$ans
@@ -311,6 +311,7 @@
   sparcv9)       asmarch=sparcv8_micro; pretty=UltraSparc ;;
   m68k)          asmarch=$arch;         pretty="MC680x0, x>=2" ;;
   i?86)          asmarch=ix86 ;         pretty=$arch ;;
+  athlon)        asmarch=ix86 ;         pretty=Athlon ;;
   hppa)          asmarch=none ;         pretty=HP
         case "$osname-`uname -r`" in
         hpux-?.09.*)  asmarch=none  ;;
