--- pari-2.5.5/config/get_readline.orig	2013-11-13 12:12:47.148846810 +0000
+++ pari-2.5.5/config/get_readline	2013-11-13 12:14:11.926846829 +0000
@@ -98,13 +98,6 @@
     if test -n "$try"; then
       rl_ncurses=$ncurses
       need_tgetent=
-    else # give termcap a try
-      pth="$with_readline_lib $libpth"
-      lib=termcap; . ./locatelib
-      if test -n "$try"; then
-        rl_termcap=$termcap
-        need_tgetent=
-      fi
     fi
     pth=$opth
   fi
