--- pari-2.3.5/config/get_readline~	2008-03-31 13:43:59.000000000 +0200
+++ pari-2.3.5/config/get_readline	2010-06-12 18:51:23.870317635 +0200
@@ -120,12 +120,6 @@
     if test -n "$try"; then
       rl_ncurses=$ncurses
       need_tgetent=
-    else # give termcap a try
-      lib=termcap; . ./locatelib
-      if test -n "$try"; then
-        rl_termcap=$termcap
-        need_tgetent=
-      fi
     fi
   fi
 # Readline -- CFLAGS
