--- pari-2.5.3/Math-Pari-2.01080605/Pari.xs.orig	2010-09-11 23:33:30.000000000 +0200
+++ pari-2.5.3/Math-Pari-2.01080605/Pari.xs	2012-10-13 09:59:33.443018028 +0200
@@ -547,7 +547,7 @@
 		      && lgef(x)==4		/* 2 terms */
 		      && (gcmp0((GEN)x[2]))	/* Free */
 		      && (gcmp1((GEN)x[3]))) { /* Leading */
-		      s = varentries[ordvar[varn(x)]]->name;
+		      s = varentries[varn(x)]->name;
 		      goto repeat;
 		  }
 		  goto ignore;
@@ -765,8 +765,8 @@
 }
 
 
-PariOUT perlOut={svputc, svputs, svOutflush, NULL};
-PariOUT perlErr={svErrputc, svErrputs, svErrflush, svErrdie};
+PariOUT perlOut={svputc, svputs, svOutflush};
+PariOUT perlErr={svErrputc, svErrputs, svErrflush};
 
 static GEN
 my_ulongtoi(ulong uv)
@@ -934,7 +934,7 @@
 	     || ((2 * sizeof(long) == sizeof(IV)) && !HAVE_UVs && in[2] < 0) )
 	    goto do_nv;
 	uv = in[2];
-	uv = (uv << TWOPOTBYTES_IN_LONG) + in[3];
+	uv = (uv << (TWOPOTBITS_IN_LONG-3)) + in[3];
 	break;
     default:
 	goto do_nv;
@@ -1025,7 +1025,7 @@
 #endif
 
 #ifndef pariK1
-#  define pariK1 (0.103810253/(BYTES_IN_LONG/4))  /* log(10)/(SL*log(2))   */
+#  define pariK1 (0.103810253/(sizeof(long)/4))  /* log(10)/(SL*log(2))   */
 #endif
 
 long
@@ -1317,17 +1324,6 @@
     moveoffstack_newer_than((SV *) GENfirstOnStack);
 }
 
-UV
-allocatemem(UV newsize)
-{
-    if (newsize) {
-	detach_stack();
-	parisize = allocatemoremem(newsize);
-	perlavma = sentinel = avma;
-    }
-    return parisize;
-}
-
 
 GEN
 callPerlFunction(entree *ep, ...)
@@ -3459,8 +3455,8 @@
 	     else
 		 warn("`%s' is an obsolete Pari function name", name);
 	 }
-	 if (ep && (EpVALENCE(ep) < EpUSER 
-		    /* && ep>=fonctions && ep < fonctions+NUMFUNC) */)) {
+	 if (ep /* && (EpVALENCE(ep) < EpUSER 
+		    && ep>=fonctions && ep < fonctions+NUMFUNC)) */) {
 	     /* Builtin */
 	   IV table_valence = 99;
 
@@ -3999,17 +3998,5 @@
 # and it needs to convert *all* the branches.
 #/* #if defined(PERL_VERSION) && (PERL_VERSION >= 6)*//* 5.6.0 has UV in the typemap */
 
-#if 0
-#UV
-#allocatemem(newsize = 0)
-#UV newsize
-
-#else	/* !( HAVE_UVs ) */
-
-unsigned long
-allocatemem(newsize = 0)
-    unsigned long newsize
-
-#endif	/* !( HAVE_UVs ) */
 
 long
--- pari-2.5.1/Math-Pari-2.01080605/func_codes.h.orig	2004-02-01 10:27:30.000000000 +0100
+++ pari-2.5.1/Math-Pari-2.01080605/func_codes.h	2012-02-19 21:26:18.835564839 +0100
@@ -587,3 +587,4 @@
   return PERL_constant_NOTFOUND;
 }
 
+extern entree functions_basic[];
